core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/entityQuery/AllEntitiesStrategy.scala

```diff
@@ -14,15 +17,22 @@ class AllEntitiesStrategy(override val repository: CompactEntityRepository,
                           entityType: String,
                           entityQuery: EntityQuery,
                           unfilteredCount: Int
+)(implicit
+  executionContext: ExecutionContext
 ) extends EntityQueryStrategy {
 
   override def getCountAndSource: Future[CountAndSource] =
-    // there is no filter so we can just use the unfiltered count
-    Future.successful(
+    EntityUtils.retryWithSortMemory[Seq[Entity]](repository.dataSource,
+                                                 isolationLevel = TransactionIsolation.ReadCommitted
+    ) {
+      repository.queries.queryEntitiesWithNoFilter(workspaceId, entityType, entityQuery)
+    } map { sourceQueryMaterializedResult =>
+      val source = Source(sourceQueryMaterializedResult)
```

Comment:
here you can see the ugliness w/r/t streaming ... we run a query, materialize its result, and then turn the result back into a stream.